File: src/main/java/ua/nure/fedorenko/kidstim/model/entity/Assignment.java
package ua.nure.fedorenko.kidstim.model.entity;

import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "assignments")
public class Assignment {
    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid")
    @Column(name = "id")
    private String id;
    @OneToOne
    @JoinColumn(name = "day")
    private Day day;
    @OneToMany
    @JoinColumn(name = "task_id")
    private List<Task> tasks;
    @OneToMany
    @JoinColumn(name = "schedule_id")
    private Schedule schedule;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Day getDay() {
        return day;
    }

    public void setDay(Day day) {
        this.day = day;
    }

    public List<Task> getTasks() {
        return tasks;
    }

    public void setTasks(List<Task> tasks) {
        this.tasks = tasks;
    }

    public Schedule getSchedule() {
        return schedule;
    }

    public void setSchedule(Schedule schedule) {
        this.schedule = schedule;
    }
}

File: src/main/java/ua/nure/fedorenko/kidstim/model/entity/Assignment.kt
package ua.nure.fedorenko.kidstim.model.entity

import org.hibernate.annotations.GenericGenerator
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "assignments")
data class Assignment(
        @Id
        @GeneratedValue(generator = "uuid")
        @GenericGenerator(name = "uuid", strategy = "uuid")
        @Column(name = "id")
        var id: String? = null,
        @OneToOne
        @JoinColumn(name = "day")
        var day: Day? = null
) : Serializable {

    @OneToMany
    @JoinColumn(name = "task_id")
    var tasks: List<Task>? = ArrayList()
    @ManyToOne
    @JoinColumn(name = "schedule_id")
    var schedule: Schedule? = null

}